---
title: "Evaluating Winner's Curse correction methods with a simulation study"
author: ""
date: ""
output: 
  html_document:
    toc: yes

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(parallel)
library(ggplot2)
library(RColorBrewer)
library(kableExtra)
library(expm)
col <- brewer.pal(8,"Dark2")
col1 <- brewer.pal(11,"RdYlBu")
col2 <- brewer.pal(11,"PRGn")
```


# 1) Independent SNPs

## Introduction

Our simulation study follows a factorial design in which GWAS summary statistics are simulated for a trait under 24 different genetic architectures, described by combinations of four parameters: 

1. sample size $n$ - `n_samples`
2. heritability $h^2$ - `h2`
3. polygenicity (proportion of effect SNPs) $\pi$ - `prop_effect`
4. selection coefficient $S$ - `S`

Our simulation strategy entails first obtaining the expected standard error for the true effect size of each SNP and subsequently, sampling an estimated effect size based on this expected standard error. This provides an estimated effect size, $\hat\beta_i$ and corresponding standard error, $\hat{\text{se}(\hat\beta_i)}$ for SNP $i, i=1,...,N$. Descriptions of the functions used to simulate summary statistics and employed in other important aspects of this study may be found in `useful_funs.R`.

Throughout this article, a fixed array of $N=1,000,000$ **independent** SNPs is assumed. Our primary focus is on evaluating methods in the setting of a quantitative trait with a normal effect size distribution. However, we also look at a quantitative trait with a bimodal effect size distribution, a quantitative trait with a skewed distribution and a binary trait with a normal effect size distribution. In the first setting, 100 sets of summary statistics are simulated for each of these 24 different genetic architectures while for the other settings, only 10 data sets are simulated in order to reduce computational time. Furthermore, the different variations of the empirical Bayes methods have not been evaluated in these other settings. 

The Winner’s Curse correction methods are applied to each data set using the R package `winnerscurse`, producing adjusted estimated effect sizes, $\hat{\beta_{\text{adj},i}}$, for each SNP $i, i=1,...,N$. The performance of these methods are investigated at two different significance thresholds, namely $\alpha_1 = 5 \times 10^{-8}$ and $\alpha_2 = 5 \times 10^{-4}$, with a stronger focus given to the more commonly used genome-wide significance threshold of $\alpha_1 = 5 \times 10^{-8}$.

In order to assess each method’s ability at producing less biased SNP-trait association estimates, various evaluation metrics were computed for each data set and method:

1. The change in MSE of significant SNPs due to method implementation - `mse`
2. The change in RMSE of significant SNPs due to method implementation - `rmse``
3. The *relative* change in  MSE of significant SNPs due to method implementation - `rel_mse`
4. The fraction of significant SNPs in which their association estimates have been improved due to method implementation - `flb`

**Note:** These metric are only computed in those simulations in which at least one significant SNP was detected. 

Letting $N_{sig}$ denote the number of significant SNPs in a particular simulated set of summary staitistics, i.e. the number of SNPs which satisfy $|z_i| > c$ with $z_i = \frac{\hat\beta_i}{\hat{\text{se}(\hat\beta_i)}}$, $c = \Phi^{-1}(1-\frac{\alpha}{2})$ and $\alpha \in \{\alpha_1,\alpha_2\}$, the change in MSE of significant SNPs may be defined as: 

$$\frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_{\text{adj,}i} - \beta_i)^2 - \frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_i - \beta_i)^2.$$
Here, $\hat\beta_i$ is the estimated naive effect size of SNP $i$, $\beta_i$ is its true effect size and $\hat\beta_{\text{adj,}i}$ is its new effect size estimate obtained as a result of application of the Winner's Curse adjustment method of interest.

Similarly, the change in RMSE of significant SNPs is defined as: $$\sqrt{\frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_{\text{adj,}i} - \beta_i)^2} - \sqrt{\frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_i - \beta_i)^2}$$ and the relative change in MSE of significant SNPs as: $$\frac{\frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_{\text{adj,}i} - \beta_i)^2 - \frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_i - \beta_i)^2}{\frac{1}{N_{sig}} \sum^{N_{sig}}_{i=1} (\hat\beta_i - \beta_i)^2}.$$ 

Finally, the fraction of $N_{sig}$ significant SNPs in which their association estimates have been improved due to method implementation may be mathematically described as: $$\frac{1}{N_{sig}} \; \sum_{i=1}^{N_{sig}}\mathbb{I} \left\{ \left| \hat\beta_i - \beta_i \right|  >  \left|\hat\beta_{\text{adj,}i} - \beta_i\right| \right\}.$$

 
The 24 different genetic architectures that will be investigated are detailed below:

```{r, echo=FALSE}

params_1 <- data.frame(Scenario=c(1:12),n_samples = c(rep(c("30,000","300,000"),6)), h2 = c(rep(c(0.3,0.3,0.8,0.8),3)), prop_effect=c(0.01,0.01,0.01,0.01,0.001,0.001,0.001,0.001,0.01,0.01,0.01,0.01), S=c(rep(-1,8),rep(0,4)))
params_2 <- data.frame(Scenario=c(13:24),n_samples = c(rep(c("30,000","300,000"),6)), h2 = c(rep(c(0.3,0.3,0.8,0.8),3)), prop_effect=c(0.001,0.001,0.001,0.001,0.01,0.01,0.01,0.01,0.001,0.001,0.001,0.001), S=c(rep(0,4),rep(1,8)))

params_1 %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE, position="float_left") %>%
  row_spec(row = 0,bold = TRUE)  

params_2 %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="left") %>%
  row_spec(row = 0,bold = TRUE)



```

$~$
$~$
$~$


**Note:** In the description and visualisation of results throughout this document we will regularly omit the settings with selection coefficient $S$ equal to 1 or -1 and just include those with the selection coefficient $S$ equal to zero. The reason for this being ease of interpretation of results. It was noted that altering the selection coefficient did not seem to make much of a change to the sets of summary statistics simulated or to the results obtained upon application of the Winner's Curse correction methods.


## Preliminary investigation 

In this section, we focus on a quantitative trait with a normal effect size distribution and investigate a number of different statistics with respect to the 24 genetic architectures described above. First, in order to gain a better insight into what patterns may be present in the sets of summary statistics, we simulate a single set of GWAS summary statistics and plot $z$ vs $\text{bias}$  in which $\text{bias} = \hat\beta - \beta$ for each of the 8 scenarios with selection coefficient equal to zero. On all figures, the <span style="color: red;"> bright red </span> line corresponds to the significance threshold of $5 \times 10^{-8}$ while the <span style="color: darkred;"> darker red </span> line relates to $5 \times 10^{-4}$. The points corresponding to SNPs which are *significantly* overestimated and are significant at a threshold of $5 \times 10^{-4}$ are coloured in <span style="color: navy;"> navy </span>. A SNP is determined to be *significantly* overestimated if it satisfies $\left| \hat\beta_{i} \right| > \left| \beta_{i} \right| + 1.96\cdot\sigma_{i}$. In this instance, $\sigma_i = \hat{\text{se}(\hat\beta_i)}$.

$~$
$~$
$~$
$~$

```{r, echo=FALSE}
# required functions!
n_snps <- 10^6

simulate_ss <- function(H,Pi,nid,sc){
  effect_snps <- Pi*n_snps
  maf <- runif(n_snps,0.01,0.5)
  true_beta <- rnorm(effect_snps,0,sd=sqrt((2*maf*(1-maf))^sc)) 
  var_y <- sum(2*maf[1:effect_snps]*(1-maf[1:effect_snps])*true_beta^2)/H
  true_beta <- true_beta/sqrt(var_y) 
  true_beta <- c(true_beta, rep(0,n_snps-effect_snps))
  se <- sqrt((1 - 2*maf*(1-maf)*true_beta^2)/(2*(nid-2)*maf*(1-maf)))
  stats <- data.frame(true_beta,se,maf)
  return(stats)
}

simulate_est <- function(stats){
  est <- data.frame(rsid=seq(1,n_snps),beta=rnorm(n=n_snps,mean=stats$true_beta,sd=stats$se),se=stats$se)
  return(est)
}

simulate_ss_bim <- function(H,Pi,nid,sc){
  effect_snps <- Pi*n_snps
  maf <- runif(n_snps,0.01,0.5)
  true_beta <- c(rnorm(0.5*effect_snps,2.5,sd=sqrt((2*maf*(1-maf))^sc)),rnorm(0.5*effect_snps,0,sd=sqrt((2*maf[(0.5*effect_snps+1):effect_snps]*(1-maf[(0.5*effect_snps+1):effect_snps]))^sc)))
  var_y <- sum(2*maf[1:effect_snps]*(1-maf[1:effect_snps])*true_beta^2)/H
  true_beta <- true_beta/sqrt(var_y)
  true_beta <- c(true_beta, rep(0,n_snps-effect_snps))
  se <- sqrt((1 - 2*maf*(1-maf)*true_beta^2)/(2*(nid-2)*maf*(1-maf)))
  stats <- data.frame(true_beta,se)
  return(stats)
}

## convert logit values to probabilities
expit <- function(x){exp(x)/(1+exp(x))}

## find Gamma0 for logistic model so prevalence is 0.1
myfunc <- function(MAF,Gamma0, Gamma1, prev=0.1){
  MAF^2*expit(Gamma0+Gamma1*2)+2*MAF*(1-MAF)*expit(Gamma0+Gamma1)+(1-MAF)^2*expit(Gamma0)-prev
}

## calculate asymptotic version of (X^t W X)
asymp_var_logistic <- function(n,G_prob,Gamma_0,Gamma_1){
  N <- length(Gamma_0)
  disease_probs <- expit(outer(Gamma_0,c(1,1,1)) +  outer(Gamma_1,c(0,1,2)))
  diag_weights <- disease_probs*(1-disease_probs)
  a <- n*apply(G_prob*diag_weights,1,sum)
  b <- n*apply(G_prob*diag_weights*matrix(rep(c(0,1,2),N),byrow=TRUE,nrow=N),1,sum)
  c <-  b
  d <- n*apply(G_prob*diag_weights*matrix(rep(c(0,1,4),N),byrow=TRUE,nrow=N),1,sum)
  ##  invert matrix and take element for Gamma1
  return(a/(a*d-b*c))

}

## Grid-like computation of beta_0 corresponding to values for maf and true_beta
## with fixed prevalence of 0.1 saves computational time
params <- expand.grid(
  maf = seq(0.01,0.5,by=0.01),
  true_beta = seq(-1.5,1.5,by=0.01)
)
params$maf <- round(params$maf,2)
params$true_beta <- round(params$true_beta,2)
params$Gamma_0 <- c(rep(0,nrow(params)))
for(i in 1:nrow(params)) {params$Gamma_0[i] <- uniroot(myfunc,Gamma1=params$true_beta[i],MAF=params$maf[i],prev=0.1,lower=-10, upper=10)$root}


simulate_ss_bin <- function(H2,Pi,nid,sc){
  effect_snps <- Pi*n_snps
  maf <- runif(n_snps,0.01,0.5)
  true_beta <- rnorm(effect_snps,0,sd=sqrt((2*maf*(1-maf))^sc))
  scaling <- (1.6^2*H2)/((sum(2*maf[1:effect_snps]*(1-maf[1:effect_snps])*true_beta^2))*(1-H2))
  true_beta <- true_beta*sqrt(scaling)
  true_beta <- c(true_beta, rep(0,n_snps-effect_snps))

  rounded_maf <- round(maf,2)
  rounded_true_beta <- round(true_beta,2)
  Gamma0 <- c(rep(0,n_snps))

  dat <- data.frame(rounded_maf, rounded_true_beta)
  for(i in 1:n_snps) Gamma0[i] <- params$Gamma_0[7500 + 5000*rounded_true_beta[i] + 100*rounded_maf[i]]

  ##  assuming HWE
  G_prob <- cbind((1-maf)^2,2*maf*(1-maf),maf^2)
  Gamma1 <- true_beta
  var_Gamma_y <- asymp_var_logistic(nid,G_prob,Gamma0,Gamma1)
  se <- sqrt(var_Gamma_y)
  stats <- data.frame(true_beta,se)
  return(stats)
}


simulate_ss_exp <- function(H,Pi,nid,sc,rep_nid=1){
  effect_snps <- Pi*n_snps
  maf <- runif(n_snps,0.01,0.5)
  true_beta <- c(-rexp(n=.1*effect_snps,rate = 1/sqrt((2*maf*(1-maf))^sc)), rexp(n=.9*effect_snps,rate = 1/sqrt((2*maf[(0.1*effect_snps+1):effect_snps]*(1-maf[(0.1*effect_snps+1):effect_snps]))^sc))) 
  var_y <- sum(2*maf[1:effect_snps]*(1-maf[1:effect_snps])*true_beta^2)/H
  true_beta <- true_beta/sqrt(var_y)
  true_beta <- c(true_beta, rep(0,n_snps-effect_snps))
  se <- sqrt((1 - 2*maf*(1-maf)*true_beta^2)/(2*(nid-2)*maf*(1-maf)))
  rep_se <- sqrt((1 - 2*maf*(1-maf)*true_beta^2)/(2*(rep_nid*nid-2)*maf*(1-maf)))
  stats <- data.frame(true_beta,se,rep_se)
  return(stats)
}

```


```{r, warning=FALSE,echo=FALSE,fig.show="hold", out.width="50%"}
set.seed(1998)

par(mar = c(4, 4, .1, .1))

## Scenario 9
ss <- simulate_ss(H=0.3,Pi=0.01,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + xlim(-6, 6) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 9", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[1]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 



## Scenario 10
ss <- simulate_ss(H=0.3,Pi=0.01,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 10", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[2]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 11
ss <- simulate_ss(H=0.8,Pi=0.01,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 11", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[3]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 12
ss <- simulate_ss(H=0.8,Pi=0.01,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 12", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[4]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 13
ss <- simulate_ss(H=0.3,Pi=0.001,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 13", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[5]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 



## Scenario 14
ss <- simulate_ss(H=0.3,Pi=0.001,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 14", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[6]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 



## Scenario 15
ss <- simulate_ss(H=0.8,Pi=0.001,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 15", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[7]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 16
ss <- simulate_ss(H=0.8,Pi=0.001,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 16", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[8]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


```



Next, over 100 simulations, we look at the number of significant SNPs, the proportion of these SNPs that have association estimates more extreme than their true effect size, the proportion of these SNPs which are *significantly* overestimated and the MSE of significant SNPs at two different thresholds; the common genome-wide significance threshold of $5 \times 10^{-8}$ and a higher threshold of $5 \times 10^{-4}$. Large values for the proportion of *significantly* overestimated SNPs and the MSE of significant SNPs provide evidence for a great degree of bias induced by Winner's Curse. 

Running the code provided in `nsig_prop_bias_100sim.R`, we obtain the following results:

```{r, echo=FALSE}

results_nsig_5e_8 <- read.csv("results/norm_nsig_prop_bias_5e_8.csv")
results_nsig_5e_4 <- read.csv("results/norm_nsig_prop_bias_5e_4.csv")

results_nsig <- data.frame(Scenario=c(1:24),n_samples=c(rep(c("30,000","300,000"),12)),results_nsig_5e_8[3:5], nsig_5e_8=round(results_nsig_5e_8$n_sig,2), pb58 = round(results_nsig_5e_8$prop_bias,4), px58 = round(results_nsig_5e_8$prop_x,4), mse58 = round(results_nsig_5e_8$mse,6), nsig_5e_4=round(results_nsig_5e_4$n_sig,2), pb54 = round(results_nsig_5e_4$prop_bias,4), px54 = round(results_nsig_5e_4$prop_x,4), mse54 = round(results_nsig_5e_4$mse,6), nsig_5e_8_sd=round(results_nsig_5e_8$n_sig_sd,3), pb58sd=round(results_nsig_5e_8$prop_bias_sd,4), px58sd=round(results_nsig_5e_8$prop_x_sd,4), mse58sd=round(results_nsig_5e_8$mse_sd,6),  nsig_5e_4_sd=round(results_nsig_5e_4$n_sig_sd,3),  pb54sd=round(results_nsig_5e_4$prop_bias_sd,4), px54sd=round(results_nsig_5e_4$prop_x_sd,4), mse54sd=round(results_nsig_5e_4$mse_sd,6))

names(results_nsig)[names(results_nsig) == "nsig_5e_8"] <- "n_sig 5e-8"
names(results_nsig)[names(results_nsig) == "nsig_5e_8_sd"] <- "sd(n_sig) 5e-8"
names(results_nsig)[names(results_nsig) == "pb58"] <- "prop_bias 5e-8"
names(results_nsig)[names(results_nsig) == "pb58sd"] <- "sd(prop_bias) 5e-8"
names(results_nsig)[names(results_nsig) == "px58"] <- "prop_x 5e-8"
names(results_nsig)[names(results_nsig) == "px58sd"] <- "sd(prop_x) 5e-8"
names(results_nsig)[names(results_nsig) == "nsig_5e_4"] <- "n_sig 5e-4"
names(results_nsig)[names(results_nsig) == "nsig_5e_4_sd"] <- "sd(n_sig) 5e-4"
names(results_nsig)[names(results_nsig) == "pb54"] <- "prop_bias 5e-4"
names(results_nsig)[names(results_nsig) == "pb54sd"] <- "sd(prop_bias) 5e-4"
names(results_nsig)[names(results_nsig) == "px54"] <- "prop_x 5e-4"
names(results_nsig)[names(results_nsig) == "px54sd"] <- "sd(prop_x) 5e-4"
names(results_nsig)[names(results_nsig) == "mse58"] <- "mse 5e-8"
names(results_nsig)[names(results_nsig) == "mse58sd"] <- "sd(mse) 5e-8"
names(results_nsig)[names(results_nsig) == "mse54"] <- "mse 5e-4"
names(results_nsig)[names(results_nsig) == "mse54sd"] <- "sd(mse) 5e-4"


results_nsig %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FAE5D3") %>%
  column_spec(2,background="#FDF2E9") %>%
  column_spec(3,background="#FDF2E9") %>%
  column_spec(4,background="#FDF2E9") %>%
  column_spec(5,background="#FDF2E9", extra_css="border-right: 1px solid") %>% 
  column_spec(9, extra_css="border-right: 1px solid") %>%
  column_spec(13, extra_css="border-right: 1px solid") %>%
   column_spec(14, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(15, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(17, background = "#F5F5F5", italic=TRUE, extra_css="border-right: 1px solid") %>%
  column_spec(18, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(16, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(19, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(20, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(21, background = "#F5F5F5", italic=TRUE) %>%
  scroll_box(width = "100%", height = "300px")

```

$~$
$~$
$~$


We show how the number of significant SNPs differs for the 8 different scenarios with $S=0$ at the significance threshold $5 \times 10^{-8}$:

```{r, echo=FALSE}

results_nsig_5e_8_all <- read.csv("results/norm_nsig_prop_bias_5e_8_all.csv")


data_neutralS <- results_nsig_5e_8_all[which(results_nsig_5e_8_all$S==0),]
data_neutralS$scenario <- c(rep("Scenario 09",100), rep("Scenario 10",100), rep("Scenario 11",100), rep("Scenario 12",100), rep("Scenario 13",100), rep("Scenario 14",100), rep("Scenario 15",100), rep("Scenario 16",100))
data_neutralS$n_samples <- c(rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100))
data_neutralS <- data_neutralS[data_neutralS$n_sig != 0,]


ggplot(data_neutralS,aes(x=n_samples,y=n_sig,colour=scenario)) + geom_boxplot(position=position_dodge(0.2),size=0.5,aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + facet_grid(h2~prop_effect,labeller=label_both) + xlab("Sample size") + ylab("No. sig SNPs") + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())

```



We can also visualise the proportion of SNPs that are *significantly* overestimated at the significance threshold of $5 \times 10^{-8}$ as follows: 

```{r, echo=FALSE}
ggplot(data_neutralS,aes(x=n_samples,y=prop_x,colour=scenario)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + xlab("Sample size")+ facet_grid(h2~prop_effect,labeller=label_both) + ylab(expression(paste("Prop. sig SNPs", italic(" significantly "), "overestimated"))) + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())

ggplot(data_neutralS,aes(x=n_sig,y=prop_x,colour=scenario)) + geom_point(aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + xlab("No. sig SNPs")+ ylab(expression(paste("Prop. sig SNPs", italic(" significantly "), "overestimated"))) + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())
```



**Discussion:**

<span class="star">&#x2605;</span>   It is important to note here that for scenarios 1, 9 and 17, very few significant SNPs are detected on average. In some instances, we may even find that no SNPs are deemed significant at a threshold of $5 \times 10^{-8}$. We must keep this observation in mind going forward as we investigate the performance of methods under these three scenarios.   

For both thresholds, the average number of significant SNPs increases as sample size increases, as expected. It also increases with heritability. However, the effect of changing `prop_effect` is more interesting. Decreasing the proportion of effect SNPs from 0.01 to 0.001 results in the number of significant SNPs increasing for a sample size of 30,000 while we witness the number of SNPs passing the genome-wide significance threshold decreasing for a larger sample size of 300,000. 

Furthermore, increasing sample size and increasing heritability from 0.3 to 0.8 all tend to decrease the fraction of significant SNPs whose estimates are more extreme than their true effect size. Decreasing polygenicity from 0.01 to 0.001 also has this same effect at a significance threshold of $5 \times 10^{-8}$.

As sample size increases from 30,000 to 300,000 in all scenarios, `prop_x`, the proportion of significant SNPs which have been found to be significantly overestimated, decreases. This is a strong indicator that as the value of `n_samples` increases, we expect the bias induced by Winner's Curse to be less of a problem among significant SNPs. However, this could also be due to the fact that as sample size increases, the number of significant SNPs passing the genome-wide significance threshold of `5e-8` also increases. This observation is also evident from the $z$ vs $\text{bias}$ plots in which we can see much smaller values of bias for the greater sample size of 300,000. 



## Quantitative trait with normal effect size distribution

For this quantitative trait with a normal effect size distribution, using the code detailed in `winnerscurse_sims_ind.R`, we evaluated ten different Winner's Curse methods across each of the 24 scenarios using the four previously detailed evaluation metrics. These methods include the three conditional likelihood methods, FDR Inverse Quantile Transformation `FIQT`, the bootstrap method `boot` as well as five variations of the empirical Bayes method, namely `EB`, `EB_df`, `EB_scam`, `EB_gam_po` and `EB_gam_nb`. 

All results of the simulations are plotted in this instance for the scenarios in which $S$ is fixed at 0. These figures allow us to see more clearly the scenarios in which it would be beneficial to apply a Winner's Curse correction method and also, provide us with a better indication of which method we should use.

A replication method which selects significant SNPs from a discovery GWAS and then uses a replication GWAS of the *same size* to obtain association estimates for these SNPs is also included in the plots. This can be viewed as acting as a form of benchmark for the other methods. 

Rankings, one for each method, are also given. We first look at the case when the significance threshold is fixed at $5 \times 10^{-8}$ and then, follow this by analysing the results when the significance threshold is changed to a lower one of $5 \times 10^{-4}$. For the greater threshold of $5 \times 10^{-4}$, only the first evaluation metric is considered.

$~$ $~$

**Evaluation metric 1 with threshold** $5 \times 10^{-8}$:

Summary of results for `mse` contained in `norm_5e-8_100sim_ave.csv`:

```{r,echo=FALSE}

norm_5e_8_100sim <- read.csv("results/norm_5e-8_100sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = norm_5e_8_100sim$h2[1:24], prop_effect = norm_5e_8_100sim$prop_effect[1:24], S= norm_5e_8_100sim$S[1:24], EB = round(norm_5e_8_100sim$mse[1:24],6), EB_df = round(norm_5e_8_100sim$mse[(1+24):(24+24)],6), EB_scam = round(norm_5e_8_100sim$mse[(1+2*24):(24+2*24)],6), EB_gam_po = round(norm_5e_8_100sim$mse[(1+3*24):(24+3*24)],6), EB_gam_nb = round(norm_5e_8_100sim$mse[(1+4*24):(24+4*24)],6), FIQT = round(norm_5e_8_100sim$mse[(1+5*24):(24+5*24)],6), boot = round(norm_5e_8_100sim$mse[(1+6*24):(24+6*24)],6), cl1 = round(norm_5e_8_100sim$mse[(1+7*24):(24+7*24)],6), cl2 = round(norm_5e_8_100sim$mse[(1+8*24):(24+8*24)],6), cl3 = round(norm_5e_8_100sim$mse[(1+9*24):(24+9*24)],6), rep = round(norm_5e_8_100sim$mse[(1+10*24):(24+10*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:16]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

## plotting full data set! 
norm_5e_8_100sim_all <- read.csv("results/norm_5e-8_100sim_all.csv")

data_neutralS <- norm_5e_8_100sim_all[which(norm_5e_8_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic")) 


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$

**Evaluation metric 2 with threshold** $5 \times 10^{-8}$:

Summary of results for `rmse` contained in `norm_5e-8_100sim_ave.csv`:

```{r,echo=FALSE}
norm_5e_8_100sim <- read.csv("results/norm_5e-8_100sim_ave.csv")
results_rmse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = norm_5e_8_100sim$h2[1:24], prop_effect = norm_5e_8_100sim$prop_effect[1:24], S= norm_5e_8_100sim$S[1:24], EB = round(norm_5e_8_100sim$rmse[1:24],6), EB_df = round(norm_5e_8_100sim$rmse[(1+24):(24+24)],6), EB_scam = round(norm_5e_8_100sim$rmse[(1+2*24):(24+2*24)],6), EB_gam_po = round(norm_5e_8_100sim$rmse[(1+3*24):(24+3*24)],6), EB_gam_nb = round(norm_5e_8_100sim$rmse[(1+4*24):(24+4*24)],6), FIQT = round(norm_5e_8_100sim$rmse[(1+5*24):(24+5*24)],6), boot = round(norm_5e_8_100sim$rmse[(1+6*24):(24+6*24)],6), cl1 = round(norm_5e_8_100sim$rmse[(1+7*24):(24+7*24)],6), cl2 = round(norm_5e_8_100sim$rmse[(1+8*24):(24+8*24)],6), cl3 = round(norm_5e_8_100sim$rmse[(1+9*24):(24+9*24)],6), rep = round(norm_5e_8_100sim$rmse[(1+10*24):(24+10*24)],6))

results_rmse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in RMSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_rmse[,6:16]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in RMSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

ggplot(data_neutralSa,aes(x=n_samples,y=rmse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$



**Evaluation metric 3 with threshold** $5 \times 10^{-8}$:

Summary of results for `rel_mse` contained in `norm_5e-8_100sim_ave.csv`:

```{r,echo=FALSE}

results_rel_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = norm_5e_8_100sim$h2[1:24], prop_effect = norm_5e_8_100sim$prop_effect[1:24], S= norm_5e_8_100sim$S[1:24], EB = round(norm_5e_8_100sim$rel_mse[1:24],4), EB_df = round(norm_5e_8_100sim$rel_mse[(1+24):(24+24)],4), EB_scam = round(norm_5e_8_100sim$rel_mse[(1+2*24):(24+2*24)],4), EB_gam_po = round(norm_5e_8_100sim$rel_mse[(1+3*24):(24+3*24)],4), EB_gam_nb = round(norm_5e_8_100sim$rel_mse[(1+4*24):(24+4*24)],4), FIQT = round(norm_5e_8_100sim$rel_mse[(1+5*24):(24+5*24)],4), boot = round(norm_5e_8_100sim$rel_mse[(1+6*24):(24+6*24)],4), cl1 = round(norm_5e_8_100sim$rel_mse[(1+7*24):(24+7*24)],4), cl2 = round(norm_5e_8_100sim$rel_mse[(1+8*24):(24+8*24)],4), cl3 = round(norm_5e_8_100sim$rel_mse[(1+9*24):(24+9*24)],4), rep = round(norm_5e_8_100sim$rel_mse[(1+10*24):(24+10*24)],4))

results_rel_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#D4EFDF") %>%
  column_spec(2,background="#E9F7EF") %>%
  column_spec(3,background="#E9F7EF") %>%
  column_spec(4,background="#E9F7EF") %>%
  column_spec(5,background="#E9F7EF",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* relative change in MSE over all significant SNPs across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_rel_mse[,6:16]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Relative change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

ggplot(data_neutralSa,aes(x=n_samples,y=rel_mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
    ylab(expression(paste("Rel.", italic(" change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=rel_mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
    ylab(expression(paste("Rel.", italic(" change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))

 
```

$~$ $~$ $~$ $~$

**Evaluation metric 4 with threshold** $5 \times 10^{-8}$:

Summary of results for `flb` contained in `norm_5e-8_100sim_ave.csv`:

```{r,echo=FALSE}


results_flb <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = norm_5e_8_100sim$h2[1:24], prop_effect = norm_5e_8_100sim$prop_effect[1:24], S= norm_5e_8_100sim$S[1:24], EB = round(norm_5e_8_100sim$flb[1:24],4), EB_df = round(norm_5e_8_100sim$flb[(1+24):(24+24)],4), EB_scam = round(norm_5e_8_100sim$flb[(1+2*24):(24+2*24)],4), EB_gam_po = round(norm_5e_8_100sim$flb[(1+3*24):(24+3*24)],4), EB_gam_nb = round(norm_5e_8_100sim$flb[(1+4*24):(24+4*24)],4), FIQT = round(norm_5e_8_100sim$flb[(1+5*24):(24+5*24)],4), boot = round(norm_5e_8_100sim$flb[(1+6*24):(24+6*24)],4), cl1 = round(norm_5e_8_100sim$flb[(1+7*24):(24+7*24)],4), cl2 = round(norm_5e_8_100sim$flb[(1+8*24):(24+8*24)],4), cl3 = round(norm_5e_8_100sim$flb[(1+9*24):(24+9*24)],4), rep = round(norm_5e_8_100sim$flb[(1+10*24):(24+10*24)],4))


results_flb %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#D4E6F1") %>%
  column_spec(2,background="#EAF2F8") %>%
  column_spec(3,background="#EAF2F8") %>%
  column_spec(4,background="#EAF2F8") %>%
  column_spec(5,background="#EAF2F8",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")

```

$~$ $~$ $~$ $~$

The *mean* fraction of significant SNPs with improved association estimates due to method implementation across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_flb[,6:16]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in descending order:

```{r, echo=FALSE}
rank(-summary2)
```

$~$ $~$ $~$ $~$

Fraction of significant SNPs with improved association estimates due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

ggplot(data_neutralSa,aes(x=n_samples,y=flb,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
    ylab(expression(paste("Fraction of", italic(" improved "), "sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0.5, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=flb,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
    ylab(expression(paste("Fraction of", italic(" improved "), "sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0.5, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))



```

$~$ $~$ $~$ $~$

**Evaluation metric 1 with threshold** $5 \times 10^{-4}$:

Summary of results for `mse` contained in `norm_5e-4_100sim_ave.csv`:

```{r,echo=FALSE}
norm_5e_8_100sim <- read.csv("results/norm_5e-4_100sim_ave.csv")
results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = norm_5e_8_100sim$h2[1:24], prop_effect = norm_5e_8_100sim$prop_effect[1:24], S= norm_5e_8_100sim$S[1:24], EB = round(norm_5e_8_100sim$mse[1:24],6), EB_df = round(norm_5e_8_100sim$mse[(1+24):(24+24)],6), EB_scam = round(norm_5e_8_100sim$mse[(1+2*24):(24+2*24)],6), EB_gam_po = round(norm_5e_8_100sim$mse[(1+3*24):(24+3*24)],6), EB_gam_nb = round(norm_5e_8_100sim$mse[(1+4*24):(24+4*24)],6), FIQT = round(norm_5e_8_100sim$mse[(1+5*24):(24+5*24)],6), boot = round(norm_5e_8_100sim$mse[(1+6*24):(24+6*24)],6), cl1 = round(norm_5e_8_100sim$mse[(1+7*24):(24+7*24)],6), cl2 = round(norm_5e_8_100sim$mse[(1+8*24):(24+8*24)],6), cl3 = round(norm_5e_8_100sim$mse[(1+9*24):(24+9*24)],6), rep = round(norm_5e_8_100sim$mse[(1+10*24):(24+10*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:16]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-4}$:

```{r,echo=FALSE}
## plotting full data set! 
norm_5e_4_100sim_all <- read.csv("results/norm_5e-4_100sim_all.csv")

data_neutralS <- norm_5e_4_100sim_all[which(norm_5e_4_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSa)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$


**Discussion:**

It is worth noting from the above simulations how the Winner's Curse methods tend to break down, or equivalently, no longer make improvements based on the first evaluation metric, `rmse` when the proportion of effect SNPs is 0.001. This is a measure of polygenicity. That said, both the original empirical Bayes method and the variation of the empirical Bayes method which uses the `gam` function and a negative binomial distribution of counts perform very similar to just taking a replication estimate when the replication sample is of the same size as the discovery GWAS, i.e. `n_samples` is equivalent in both data sets. Under the assumption that SNPs are independent, this observation *strongly supports the use of empirical Bayes or empirical Bayes* `gam-nb` as Winner's Curse adjustment methods, particularly when a replication GWAS is not available. The consistency of these two methods is what makes them stand out above the other considered methods. Unlike the bootstrap, FIQT and other variations of the empirical Bayes method, it is rare that it will result in an increase in the relative MSE or MSE over all significant SNPs. For that reason, employing these methods gives us confidence that we have not adjusted our estimated effect sizes so that they are now less accurate, regardless of the underlying genetic architecture or number of samples. It is surprising to see the empirical Bayes method which uses two shape constrained additive models (SCAMs) perform extremely poorly when the sample size is 300,000 and the proportion of effect SNPs 0.001. This method was seen to perform well with real data. 

For `prop_effect = 0.01`, our proposed bootstrap method for summary statistics performs in a comparable manner to both empirical Bayes and the replication method. However, the bootstrap method ceases to perform as well at obtaining less biased association estimates when the proportion of effect SNPs is reduced to 0.001. That said, it is still seen to perform competitively with respect to other currently published methods. 

On all occasions, the conditional likelihood methods tend to perform poorly compared to the other methods. This is likely due to the fact that the conditional likelihood approach can over-correct estimated effect sizes, especially those that lie close to the significance threshold. 

Clearly, even though we also look at using a significance threshold of $5 \times 10^{-4}$, it is the genome-wide significance threshold of $5 \times 10^{-8}$ that is of most interest to us as using the higher threshold of $5 \times 10^{-4}$ often results in the detection of many false positives.  

At the $5 \times 10^{-4}$ threshold, we see that all methods result in a negative change in MSE as desired for sample sizes of 30,000. However, as mentioned above, we must bear in mind that when using this lower threshold, we risk capturing SNPs which in fact have no effect, i.e. they have a true effect size of 0 and are false positives. Thus, as all of these methods may be to some degree considered as 'shrinkage' methods, it is expected that we would see an improvement in the MSE over all significant SNPs. The approach of obtaining replication estimates clearly outperforms all methods. However, it is still the original empirical Bayes method that seems to be second best. For the larger sample size of 300,000, it is worth noting that the other four variations of the empirical Bayes method do not seem to perform well compared to other methods, with a positive value for change in MSE noted in some cases. When the proportion of effect SNPs is 0.001, `EB_gam_nb` is not behaving as well as seen previously.  



## Quantitative trait with bimodal effect size distribution

A major limitation of the evaluation of methods using simulations is that informing the simulations correctly can be very difficult. For example, not all traits will have a normally distributed true effect size distribution. Therefore, in this section, in order to see if the methods perform differently when the distribution is changed, we consider a quantitative trait with a ***bimodal*** effect size distribution. In order to create a bimodal distribution, we simulate 50% of effect sizes of the true effect SNPs from a normal distribution centered at 0 while the other half are generated from a normal distribution with mean 2.5. As mentioned previously, in this setting, only 10 sets of summary statistics are simulated for each scenario with only six Winner's Curse correction methods evaluated. The code contained in `bim_skew_bin_50sim.R` details the simulation process in this instance and the function used to simulate this bimodal distribution may be found in `useful_funs.R` with the function named `simulate_ss_bim`. 

Similar to above, to provide us with an idea of what this bimodal distribution looks like, we look at the $z$ vs $\text{bias}$ plot for Scenarios 11 and 12. 

```{r, warning=FALSE,echo=FALSE,fig.show="hold", out.width="50%"}
set.seed(1998)

par(mar = c(4, 4, .1, .1))

## Scenario 11
ss <- simulate_ss_bim(H=0.8,Pi=0.01,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 11", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[3]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.1, 0.1) 

## Scenario 12
ss <- simulate_ss_bim(H=0.8,Pi=0.01,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 12", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[4]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.1, 0.1) 

```

$~$ $~$

Next, the process of describing the results illustrated in the previous section using the first bias evaluation metric, `mse` with significance thresholds of $5 \times 10^{-8}$ and $5 \times 10^{-4}$, is repeated. 

**Evaluation metric 1 with threshold** $5 \times 10^{-8}$:

Summary of results for `mse` contained in `bim_5e-8_50sim_ave.csv`:

```{r,echo=FALSE}

bim_5e_8_10sim <- read.csv("results/bim_5e-8_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = bim_5e_8_10sim$h2[1:24], prop_effect = bim_5e_8_10sim$prop_effect[1:8], S= bim_5e_8_10sim$S[1:24], EB = round(bim_5e_8_10sim$mse[1:24],6), FIQT = round(bim_5e_8_10sim$mse[(1+24):(24+24)],6), boot = round(bim_5e_8_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(bim_5e_8_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(bim_5e_8_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(bim_5e_8_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(bim_5e_8_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

## plotting full data set! 

bim_5e_8_100sim_all <- read.csv("results/bim_5e-8_50sim_all.csv")

data_neutralS <- bim_5e_8_100sim_all[which(bim_5e_8_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$

**Evaluation metric 1 with threshold** $5 \times 10^{-4}$:

Summary of results for `mse` contained in `bim_5e-4_50sim_ave.csv`:

```{r,echo=FALSE}

bim_5e_4_10sim <- read.csv("results/bim_5e-4_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = bim_5e_4_10sim$h2[1:24], prop_effect = bim_5e_4_10sim$prop_effect[1:8], S= bim_5e_4_10sim$S[1:24], EB = round(bim_5e_4_10sim$mse[1:24],6), FIQT = round(bim_5e_4_10sim$mse[(1+24):(24+24)],6), boot = round(bim_5e_4_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(bim_5e_4_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(bim_5e_4_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(bim_5e_4_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(bim_5e_4_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-4}$:

```{r,echo=FALSE}

## plotting full data set! 

bim_5e_4_100sim_all <- read.csv("results/bim_5e-4_50sim_all.csv")

data_neutralS <- bim_5e_4_100sim_all[which(bim_5e_4_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$

**Discussion:**

At the $5 \times 10^{-8}$ significance threshold for this bimodal effect size distribution, we see that the empirical Bayes method continues to perform well, with it always seeming to follow the same pattern as the 'replication' method in the plots. Our proposed bootstrap method behaves very competitively when the proportion of true effect SNPs is 0.01. Again, the simulations suggest that the conditional likelihood methods should not be employed to assist in reducing the bias induced by winner's curse. The greater threshold shows similar results but with all change in MSE values negative for each method as anticipated at this threshold of $5 \times 10^{-4}$.


## Quantitative trait with skewed effect size distribution

Next, a quantitative trait with a ***skewed*** distribution of effect sizes is investigated. Details for the manner in which this skewed distribution is simulated may be found in `useful_funs.R` with the function named `simulate_ss_exp`. Recall that in this setting, only 10 sets of summary statistics are simulated for each scenario with only six Winner's Curse correction methods evaluated. The code contained in `bim_skew_bin_50sim.R` details this simulation process and the function used to simulate the skewed distribution may be found in `useful_funs.R` with the function named `simulate_ss_exp`. 

We look at the $z$ vs $\text{bias}$ plots for Scenarios 15 and 16. 

```{r, warning=FALSE,echo=FALSE,fig.show="hold", out.width="50%"}
set.seed(1998)

par(mar = c(4, 4, .1, .1))

## Scenario 15
ss <- simulate_ss_exp(H=0.8,Pi=0.001,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 15", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[7]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.1, 0.1) 

## Scenario 16
ss <- simulate_ss_exp(H=0.8,Pi=0.001,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 16", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[8]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.1, 0.1) 


```

$~$ $~$

Results of using the first bias evaluation metric, `mse` with significance thresholds of $5 \times 10^{-8}$ and $5 \times 10^{-4}$, are detailed below. 

**Evaluation metric 1 with threshold** $5 \times 10^{-8}$:

Summary of results for `mse` contained in `skew_5e-8_50sim_ave.csv`:

```{r,echo=FALSE}

skew_5e_8_10sim <- read.csv("results/skew_5e-8_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = skew_5e_8_10sim$h2[1:24], prop_effect = skew_5e_8_10sim$prop_effect[1:8], S= skew_5e_8_10sim$S[1:24], EB = round(skew_5e_8_10sim$mse[1:24],6), FIQT = round(skew_5e_8_10sim$mse[(1+24):(24+24)],6), boot = round(skew_5e_8_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(skew_5e_8_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(skew_5e_8_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(skew_5e_8_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(skew_5e_8_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

## plotting full data set! 

skew_5e_8_100sim_all <- read.csv("results/skew_5e-8_50sim_all.csv")

data_neutralS <- skew_5e_8_100sim_all[which(skew_5e_8_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$

**Evaluation metric 1 with threshold** $5 \times 10^{-4}$:

Summary of results for `mse` contained in `skew_5e-4_50sim_ave.csv`:

```{r,echo=FALSE}

skew_5e_4_10sim <- read.csv("results/skew_5e-4_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = skew_5e_4_10sim$h2[1:24], prop_effect = skew_5e_4_10sim$prop_effect[1:8], S= skew_5e_4_10sim$S[1:24], EB = round(skew_5e_4_10sim$mse[1:24],6), FIQT = round(skew_5e_4_10sim$mse[(1+24):(24+24)],6), boot = round(skew_5e_4_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(skew_5e_4_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(skew_5e_4_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(skew_5e_4_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(skew_5e_4_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-4}$:

```{r,echo=FALSE}

## plotting full data set! 

skew_5e_4_100sim_all <- read.csv("results/skew_5e-4_50sim_all.csv")

data_neutralS <- skew_5e_4_100sim_all[which(skew_5e_4_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$



**Discussion:**

With 10 simulations for this form of distribution of effect sizes, we see that we can make very similar conclusions to those we made when the true effect sizes were normally distributed. The empirical Bayes method performs the most consistently at improving the MSE over all significant SNPs. Concerningly, all other methods tend to lead to positive values for change in MSE for the majority of cases with only the empirical Bayes method seeming to be reliable for the $5 \times 10^{-8}$ threshold. 


## Binary trait with normal effect size distribution

Under the same 24 scenarios, we investigate simulating summary statistics for a ***binary*** trait with a normal effect size distribution and disease prevalence of 0.1 and evaluate the corresponding performance of the various Winner's Curse adjustment methods. 

Details as to how we have simulated this binary trait can be found in `bim_skew_bin_50sim.R`. However, it is worth noting that simulating a binary trait can be very computationally intensive and hence why our investigation here has been restricted to a mere total of 10 simulations for each scenario. The reason for this being that given values for the true effect size of a SNP, $\beta$ and its minor allele frequency, we must also simulate a value $\beta_0$ in order to obtain an appropriate value for $\text{se}(\hat\beta)$. Our grid-like set-up has reduced the computing time immensely but in comparison to simulating a quantitative trait, it still takes considerably longer. 

Compared to the quantitative trait above, we witness a lot less SNPs meeting the significance threshold of `5e-8` under many of the scenarios. 

Firstly, we plot $z$ vs $\text{bias}$ for the 8 different scenarios with a simulated binary trait and selection coefficient $S$ equal to zero. 

```{r, warning=FALSE,echo=FALSE,fig.show="hold", out.width="50%"}
set.seed(1998)

par(mar = c(4, 4, .1, .1))

## Scenario 9
ss <- simulate_ss_bin(H2=0.3,Pi=0.01,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + xlim(-6, 6) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 9", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[1]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)


## Scenario 10
ss <- simulate_ss_bin(H2=0.3,Pi=0.01,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 10", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[2]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)

## Scenario 11
ss <- simulate_ss_bin(H2=0.8,Pi=0.01,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 11", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[3]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)

## Scenario 12
ss <- simulate_ss_bin(H2=0.8,Pi=0.01,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 12", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[4]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)

## Scenario 13
ss <- simulate_ss_bin(H2=0.3,Pi=0.001,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 13", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[5]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)


## Scenario 14
ss <- simulate_ss_bin(H2=0.3,Pi=0.001,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 14", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[6]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)


## Scenario 15
ss <- simulate_ss_bin(H2=0.8,Pi=0.001,nid=30000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 15", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[7]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)

## Scenario 16
ss <- simulate_ss_bin(H2=0.8,Pi=0.001,nid=300000,sc=0)
out <- simulate_est(ss)
out$z <- out$beta/out$se
out$bias <- out$beta - ss$true_beta
subout <- out[(abs(out$beta) > (abs(ss$true_beta) + 1.96*ss$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1)  + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 16", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[8]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) + ylim(-0.3,0.3)

```

$~$ $~$

Results of using the first bias evaluation metric, `mse` with significance thresholds of $5 \times 10^{-8}$ and $5 \times 10^{-4}$, are detailed below. 

**Evaluation metric 1 with threshold** $5 \times 10^{-8}$:

Summary of results for `mse` contained in `bin_5e-8_50sim_ave.csv`:

```{r,echo=FALSE}

bin_5e_8_10sim <- read.csv("results/bin_5e-8_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = bin_5e_8_10sim$h2[1:24], prop_effect = bin_5e_8_10sim$prop_effect[1:8], S= bin_5e_8_10sim$S[1:24], EB = round(bin_5e_8_10sim$mse[1:24],6), FIQT = round(bin_5e_8_10sim$mse[(1+24):(24+24)],6), boot = round(bin_5e_8_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(bin_5e_8_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(bin_5e_8_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(bin_5e_8_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(bin_5e_8_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

## plotting full data set! 

bin_5e_8_10sim_all <- read.csv("results/bin_5e-8_50sim_all.csv")

data_neutralS <- bin_5e_8_10sim_all[which(bin_5e_8_10sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]


library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))
data_neutralSa <- data_neutralSa[data_neutralSa$flb != 1,]


data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$

**Evaluation metric 1 with threshold** $5 \times 10^{-4}$:

Summary of results for `mse` contained in `bin_5e-4_50sim_ave.csv`:

```{r,echo=FALSE}

bin_5e_4_10sim <- read.csv("results/bin_5e-4_50sim_ave.csv")

results_mse <- data.frame(Scenario=c(1:24),n_samples = c(rep(c("30,000","300,000"),12)), h2 = bin_5e_4_10sim$h2[1:24], prop_effect = bin_5e_4_10sim$prop_effect[1:8], S= bin_5e_4_10sim$S[1:24], EB = round(bin_5e_4_10sim$mse[1:24],6), FIQT = round(bin_5e_4_10sim$mse[(1+24):(24+24)],6), boot = round(bin_5e_4_10sim$mse[(1+2*24):(24+2*24)],6), cl1 = round(bin_5e_4_10sim$mse[(1+3*24):(24+3*24)],6), cl2 = round(bin_5e_4_10sim$mse[(1+4*24):(24+4*24)],6), cl3 = round(bin_5e_4_10sim$mse[(1+5*24):(24+5*24)],6), rep = round(bin_5e_4_10sim$mse[(1+6*24):(24+6*24)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 24 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:12]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-4}$:

```{r,echo=FALSE}

## plotting full data set! 

bin_5e_4_100sim_all <- read.csv("results/bin_5e-4_50sim_all.csv")

data_neutralS <- bin_5e_4_100sim_all[which(bin_5e_4_100sim_all$S==0),]
data_neutralS <- data_neutralS[data_neutralS$flb != 100,]

library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSb)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10,16)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11],col2[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$


**Discussion:**

For the $5 \times 10^{-8}$ significance threshold, there is an absence of data for Scenario 13 in the above plot. This is is due to the fact that across all 10 simulations, there was only one simulation in which at least one SNP passed the significance threshold. Again, it is safe to state that we can draw similar conclusions as to those we have noted previously. 

We also briefly look at the performance of methods at the lower threshold of $5 \times 10^{-4}$ for this binary trait. As one can see from the obtained plot, the results are as expected with all methods making improvements to the MSE and the empirical Bayes method being the most reliable. 






# 2) Non-independent SNPs


## Introduction

Previously, we have discussed and performed a simulation study in which we simulated various sets of GWAS summary statistics and subsequently, applied several Winner's Curse correction methods. In this study, we simulated the effect size of each SNP independently as this was the most straightforward approach. However, this is clearly not representative of the real world situation in which complicated linkage disequilibrium (LD) structures exist. Therefore, we have attempted to impose a certain LD structure on SNPs and simulate GWAS summary statistics based on this structure. Admittedly, the structure that we have chosen is very simple. That said, the following study is considered to be more reflective of the real world scenario than that which was executed previously in which SNPs were assumed to be independent. Summary statistics are reflective of a quantitative trait in which the true SNP-trait effect sizes are normally distributed.

A similar format is followed in which we simulate summary statistics for a total of $N = 1,000,000$ SNPs under 8 different genomic architectures which are described by combinations of four parameters, namely sample size $n$, heritability $h^2$, polygenicity (proportion of effect SNPs) $\pi$ and selection coefficient $S$. We have previously noted that changing the selection coefficient doesn't seem to make much of a difference to the set of summary statistics and so, our main focus is on the 8 scenarios in which the selection coefficient is equal to zero.

As before, we simulate the estimated effect size, $\hat\beta_j$ for each SNP $j$ as well as its corresponding standard error, $\hat{\text{se}(\hat\beta_j)}$.

**Description of mathematical background:**

Let us assume that the true effect sizes are $\boldsymbol{b} = b_1, ..., b_S$, in which there exist a total number of $S$ SNPs. We then let $\boldsymbol{X}$ be an $N \times S$ matrix of genotypes, assumed to be mean centred, that is for each column $j: \sum_{i=1}^NX_{i,j} = 0$. We assume that genotypes $\boldsymbol{X}$ affects the $N \times 1$ response vector $Y$ through the following linear model:

$$\boldsymbol{Y} = \boldsymbol{Xb } + \varepsilon$$

where $\varepsilon$ is a vector of independent zero mean normally distributed errors, i.e. it is assumed that $Var(\varepsilon) = \sigma^2\boldsymbol{I}$ where $\boldsymbol{I}$ denotes the $N \times N$ identity matrix.

Now, let $\boldsymbol{D} = Diag(d_1,...,d_S)$ where $d_j = \sum_{i=1}^NX^2_{i,j}$. Using the well-known identities $E(X_j^2) = \frac{\sum_{i=1}^NX^2_{i,j}}{N}$ and $var(X_j) = E(X_j^2) - (E(X_j))^2$, we have: $$d_j = N \cdot var(X_j) = N \cdot 2 \cdot \text{maf}_j(1-\text{maf}_j)$$

as each column of genotypes is assumed to be mean centred, $E(X_j) = 0$ for each $j$ and the variance of each SNP $j$, $var(X_j)$, is assumed to take the form $var(X_j) = 2 \cdot \text{maf}_j(1-\text{maf}_j)$.

With the above definition for $\boldsymbol{D}$, the regression coefficients, $\hat\beta_1,...,\hat\beta_S$ from the marginal regression of Y on each SNP $X_j$ can be computed via the vector equation: $$\hat\beta = (\boldsymbol{X^TX})^{-1}\boldsymbol{X^TY} = \boldsymbol{D}^{-1}\boldsymbol{X^TY}$$

Conditionally on the genotype matrix $X$, these estimated regression coefficients have variance-covariance matrix:

$$\boldsymbol{Cov}(\hat\beta) = \boldsymbol{D}^{-1}\boldsymbol{X^TX}\boldsymbol{D}^{-1}\sigma^2 = \boldsymbol{D}^{-1/2}\boldsymbol{D}^{-1/2}\boldsymbol{X^TX}\boldsymbol{D}^{-1/2}\boldsymbol{D}^{-1/2}\sigma^2 \approx \boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{-1/2}\sigma^2$$

where $\boldsymbol{R}$ is the $S \times S$ LD matrix of inter-genotype correlations, which should approximately equal the empirical correlation matrix $\boldsymbol{D^{-1/2}X^TXD^{-1/2}}$. Letting $\boldsymbol{SE}$ be the $S \times S$ diagonal matrix with element $j$ equal to $\text{se}(\hat\beta_j) = \frac{\sigma}{\sqrt{\sum_{i=1}^NX^2_{i,j}}}$, the final expression could instead be written as: $\boldsymbol{Cov}(\hat\beta) = (\boldsymbol{SE})^{1/2}\boldsymbol{R}(\boldsymbol{SE})^{1/2}$

Finally, these estimated associations are not necessarily unbiased for the true causal effects $b_1,...b_S$. Instead they are unbiased for: $$E(\hat\beta) = \boldsymbol{D}^{-1}\boldsymbol{X^T}E(\boldsymbol{Y}) = \boldsymbol{D}^{-1}\boldsymbol{X^T}E(\boldsymbol{Xb} + \varepsilon) = \boldsymbol{D}^{-1}\boldsymbol{X^TXb} = \boldsymbol{D}^{-1/2}\boldsymbol{D}^{-1/2}\boldsymbol{X^TX}\boldsymbol{D}^{-1/2}\boldsymbol{D}^{1/2}\boldsymbol{b} \approx \boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{1/2}\boldsymbol{b}$$

Multivariate normality of $\hat\beta$ is inherited from the assumption that $\varepsilon$ is normally distributed. In summary, conditional on the centred genotype matrix $\boldsymbol{X}$: $$\hat\beta \sim N(\boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{1/2}\boldsymbol{b}, \boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{-1/2}\sigma^2)$$

**Description of simulation method:**

Using the mathematical details provided above, the following steps were executed in order to obtain an estimated effect size, $\hat\beta_j$ and corresponding standard error, $\text{se}(\hat\beta_j)$ for SNP $j$, $j=1,...,N$. The process begins in a similar fashion as the simulation of summary statistics for independent SNPs.

1.  A value is provided for $\pi$, the proportion of the total SNPs which are truly associated with the trait in question. This determines the number of effect SNPs $K = \pi \cdot N, 1 \le K \le N$, forming the polygenic background.

2.  A minor allele frequency is attained for SNP $j$ using the uniform distribution $\text{maf}_j \sim U[0.01,0.5]$.

3.  Assuming that the true effect sizes, $b_j$ of associated SNPs follow a Gaussian distribution with mean 0, $b_j$ is sampled for SNP $j, j=1,...,K$, from the distribution $b_j \sim N(0,[2\text{maf}_j(1-\text{maf}_j)]^S)$. Recall that $S$ is the value of the selection coefficient. Non-effect SNP $j, j = K + 1,...,N$, is simply assigned the null true effect size, $b_j = 0$.

4.  Defining the heritability $h^2$ as the proportion of phenotypic variation, $var(Y)$, that is explained by all SNPs, $var(Y)$ can be computed as $var(Y) = \frac{\sum_{j=1}^K2\text{maf}_j(1-\text{maf}_j)\cdot b_j^2}{h^2}$, and following this, the true effect sizes are re-scaled giving $b_j = \frac{b_j}{\sqrt{var(Y)}}$ for $j=1,...,N$ in order to ensure a phenotype with variance 1, i.e. $\sigma = 1$.

5.  Next, using the `sample` function, $K$ random positions between 1 and $N$ are chosen for the effect SNPs and the vectors containing the values of true effect size, $b_j$ and minor allele frequency, $\text{maf}_j$ are adjusted accordingly.

6.  In order to reduce computation time, it is assumed that the same linkage disequilibrium structure exists in independent blocks of 100 SNPs. Therefore, for each block of 100 SNPs, the estimated effect sizes, $\hat\beta_j$ are simulated using: $$\hat\beta \sim N(\boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{1/2}\boldsymbol{b}, \boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{-1/2}\sigma^2)$$As we have already scaled $b_j$ to ensure the phenotype has variance 1, we have $\sigma^2 = 1$. The matrix $\boldsymbol{D}$ is a diagonal $100 \times 100$ matrix, in which $d_j = n \cdot 2 \cdot \text{maf}_j(1-\text{maf}_j)$ and thus, $\boldsymbol{D}^{-1/2}$ is a similar diagonal matrix with $d^{-1/2}_j = \frac{1}{\sqrt{n \cdot 2 \cdot \text{maf}_j(1-\text{maf}_j)}}$ and $\boldsymbol{D}^{1/2}$ has diagonal entries $d^{1/2}_j = \sqrt{n \cdot 2 \cdot \text{maf}_j(1-\text{maf}_j)}$. The challenge here is to choose a suitable matrix for $\boldsymbol{R}$, the $N \times N$ LD matrix of inter-genotype correlations. For simplicity, we have chosen $\boldsymbol{R}$ to be of the following format: $$
    \begin{equation*}
    R = 
    \begin{pmatrix}
    1 & \rho & \rho ^2 & \rho ^3 & \cdots & \rho^{99} \\
    \rho & 1 & \rho & \rho^2 & \cdots & \rho^{98} \\
    \rho^2 & \rho & 1 & \rho  & \cdots & \rho^{97} \\
    \vdots  & \vdots  & \vdots & \vdots & \ddots & \vdots  \\
    \rho^{99} & \rho^{98} & \rho^{97} & \rho^{96} & \cdots & 1 
    \end{pmatrix}
    \end{equation*}
    $$ The task is now to choose a suitable value for $\rho$. We choose a value of $\approx 0.9825$ for $\rho$ which permitted the construction of $\boldsymbol{R}$. Within each block of 100 SNPs, we then simulated values for $E(\hat\beta) = \boldsymbol{D}^{-1/2}\boldsymbol{R}\boldsymbol{D}^{1/2}\boldsymbol{b}$ and $\hat\beta$. We obtained $\hat\beta$ using the `rnorm` function with $\hat\beta = E(\hat\beta) + \boldsymbol{R}^{1/2}\boldsymbol{D}^{-1/2}\cdot$`rnorm(100)`. The standard errors for each SNP were easily obtained from the diagonal entries of $\boldsymbol{D}^{-1/2}$.

7.  The above process thus provided values for $E(\hat\beta_j)$, $\hat\beta_j$ and $\text{se}(\hat\beta_j)$ for each SNP $j=1,...,N$, in which the same LD structure described by the simple matrix $\boldsymbol{R}$ has been imposed for each independent block of 100 SNPs.

Firstly, in order to gain a better insight into the various scenarios by means of different combinations of the previously mentioned parameter, we simulate a single set of GWAS summary statistics and plot $z$ vs $\text{bias}$ in which $z = \frac{\hat\beta}{\text{se}(\hat\beta)}$ and $\text{bias} = \hat\beta - E(\hat\beta)$ for each of the 8 different scenarios when the selection coefficient, S is fixed at 0. On all figures, the [bright red]{style="color: red;"} line corresponds to the significance threshold of $5 \times 10^{-8}$ while the [darker red]{style="color: darkred;"} line relates to $5 \times 10^{-4}$. The points corresponding to SNPs which are *significantly* overestimated and are significant at a threshold of $5 \times 10^{-4}$ are coloured in [navy]{style="color: navy;"}.

```{r, echo=FALSE}
n_snps <- 10^6

## now to specify R!
x <- 0.9825

s <- function(n,x=0.9825){
  vector <- c()
  for (i in 1:n){
    vector[i] <- x^(i-1)
  }
  return(vector)
}

vec <- function(tot){
  vector <- c()
  for (i in 1:tot){
    vector <- c(vector,s(100-(i-1)))
  }
  return(vector)
}

m1 <- matrix(NA, 100, 100)
m1[lower.tri(m1, diag=TRUE)] <- vec(100)
m2 <- t(m1)
m2[lower.tri(m2, diag=TRUE)] <- vec(100)
R <- m2
R_sqrt <- sqrtm(R) ## this remains the same no matter what!

################################################################################
## Function to simulate LD - takes about 30 seconds to simulate 1 set of summary statistics:

simulate_ss_ld <- function(H,Pi,nid,sc,cormat=R,cormatsq=R_sqrt){
  effect_snps <- Pi*n_snps
  maf <- runif(n_snps,0.01,0.5)
  true_beta <- rnorm(effect_snps,0,sd=sqrt((2*maf[1:effect_snps]*(1-maf[1:effect_snps]))^sc))
  var_y <- sum(2*maf[1:effect_snps]*(1-maf[1:effect_snps])*true_beta^2)/H
  true_beta <- true_beta/sqrt(var_y)

  true_beta_all <- c(rep(0,n_snps))
  maf_re <- c(rep(0,n_snps))
  
  positions <- sample(1:n_snps, effect_snps, replace=F)

  for (i in 1:effect_snps){
    true_beta_all[positions[i]] <- true_beta[i]
    maf_re[positions[i]] <- maf[i]
    
  }
  
  vec <- seq(1,10^6)
  vec_new <- vec[! vec %in% positions]
  
  for (i in (effect_snps+1):n_snps){
    maf_re[vec_new[i-effect_snps]] <- maf[i]
  }
  

  mu_all <- numeric(n_snps)
  beta_hat_all <- numeric(n_snps)
  se_all <- numeric(n_snps)

  ## R defined outside of function as it is fixed!
  for (i in 1:10000){
    D_1 <- diag(x = 1/(sqrt(nid*2*maf_re[((i-1)*100+1):(i*100)]*(1-maf_re[((i-1)*100+1):(i*100)]))), nrow=100, ncol=100)
    D_2 <- diag(x = (sqrt(nid*2*maf_re[((i-1)*100+1):(i*100)]*(1-maf_re[((i-1)*100+1):(i*100)]))), nrow=100, ncol=100)
    A <- D_1 %*% cormat %*% D_2
    sigma <- D_1 %*% cormat %*% D_1
    B <- cormatsq %*% D_1
    mu_all[((i-1)*100+1):(i*100)] <- A %*% true_beta_all[((i-1)*100+1):(i*100)]
    beta_hat_all[((i-1)*100+1):(i*100)] <- mu_all[((i-1)*100+1):(i*100)] + ( B %*% rnorm(100))
    se_all[((i-1)*100+1):(i*100)] <- diag(D_1)
  }

  summary_stats <- data.frame(rsid=seq(1,n_snps),beta=beta_hat_all,se=se_all,true_beta=mu_all)
  return(summary_stats)
}


```

```{r, warning=FALSE,echo=FALSE,fig.show="hold", out.width="50%"}
set.seed(1998)

par(mar = c(4, 4, .1, .1))


## Scenario 1
out <- simulate_ss_ld(H=0.3,Pi=0.01,nid=30000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 1", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[1]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 2
out <- simulate_ss_ld(H=0.3,Pi=0.01,nid=300000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 2", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[2]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 

## Scenario 3
out <- simulate_ss_ld(H=0.8,Pi=0.01,nid=30000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 3", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[3]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1)

## Scenario 4
out <- simulate_ss_ld(H=0.8,Pi=0.01,nid=300000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 4", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.01   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
  geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[4]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 

## Scenario 5
out <- simulate_ss_ld(H=0.3,Pi=0.001,nid=30000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 5", subtitle="n_samples=30,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[5]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 6
out <- simulate_ss_ld(H=0.3,Pi=0.001,nid=300000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 6", subtitle="n_samples=300,000   h2=0.3   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[6]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1) 


## Scenario 7
out <- simulate_ss_ld(H=0.8,Pi=0.001,nid=30000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 7", subtitle="n_samples=30,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[7]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1)

## Scenario 8
out <- simulate_ss_ld(H=0.8,Pi=0.001,nid=300000,sc=0)
out$z <- out$beta/out$se
out$bias <- out$beta - out$true_beta
subout <- out[(abs(out$beta) > (abs(out$true_beta) + 1.96*out$se)) & (abs(out$beta/out$se) > qnorm((5e-4)/2, lower.tail=FALSE)),]
ggplot(out,aes(x=z,y=bias)) + geom_point(size=1) + geom_point(data=subout, 
             aes(x=z,y=bias), 
             color='navy',
             size=1) + xlab("z") +
  ylab("bias") + ggtitle("Scenario 8", subtitle="n_samples=300,000   h2=0.8   prop_effect=0.001   S=0") +  theme_classic() + geom_vline(xintercept=qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-8)/2, lower.tail=FALSE), colour="red", linetype="dashed",size=1) +
geom_vline(xintercept=qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) + geom_vline(xintercept=-qnorm((5e-4)/2, lower.tail=FALSE), colour="darkred", linetype="dashed",size=1) +
  geom_hline(yintercept=0) +
  theme(
    plot.title = element_text(hjust = 0.5, face="bold", colour=col[8]),
    plot.subtitle = element_text(hjust = 0.5, face="italic")
  ) +
  ylim(-0.1, 0.1)


```


$~$ $~$ 

In a similar manner to the preliminary investigation of the first part of this extensive study, over a total of 10 simulations, we look at the number of significant SNPs, the proportion of these SNPs that have association estimates more extreme than their true effect size, the proportion of these SNPs which are *significantly* overestimated and the MSE of significant SNPs at the common genome-wide significance threshold of $5 \times 10^{-8}$. Large values for the proportion of *significantly* overestimated SNPs and the MSE of significant SNPs provide evidence for a great degree of bias induced by Winner's Curse. 

Running the code provided in `nsig_prop_bias_LD_100sim.R`, we obtain the following results:

```{r, echo=FALSE}

results_nsig_5e_8 <- read.csv("results/nsig_prop_bias_5e-8_LD.csv")
results_nsig <- data.frame(Scenario=c(1:8),n_samples=c(rep(c("30,000","300,000"),4)),results_nsig_5e_8[3:5], nsig_5e_8=round(results_nsig_5e_8$n_sig,2), pb58 = round(results_nsig_5e_8$prop_bias,4), px58 = round(results_nsig_5e_8$prop_x,4), mse58 = round(results_nsig_5e_8$mse,6), nsig_5e_8_sd=round(results_nsig_5e_8$n_sig_sd,3), pb58sd=round(results_nsig_5e_8$prop_bias_sd,4), px58sd=round(results_nsig_5e_8$prop_x_sd,4), mse58sd=round(results_nsig_5e_8$mse_sd,6))

names(results_nsig)[names(results_nsig) == "nsig_5e_8"] <- "n_sig 5e-8"
names(results_nsig)[names(results_nsig) == "nsig_5e_8_sd"] <- "sd(n_sig) 5e-8"
names(results_nsig)[names(results_nsig) == "pb58"] <- "prop_bias 5e-8"
names(results_nsig)[names(results_nsig) == "pb58sd"] <- "sd(prop_bias) 5e-8"
names(results_nsig)[names(results_nsig) == "px58"] <- "prop_x 5e-8"
names(results_nsig)[names(results_nsig) == "px58sd"] <- "sd(prop_x) 5e-8"
names(results_nsig)[names(results_nsig) == "mse58"] <- "mse 5e-8"
names(results_nsig)[names(results_nsig) == "mse58sd"] <- "sd(mse) 5e-8"


results_nsig %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FAE5D3") %>%
  column_spec(2,background="#FDF2E9") %>%
  column_spec(3,background="#FDF2E9") %>%
  column_spec(4,background="#FDF2E9") %>%
  column_spec(5,background="#FDF2E9", extra_css="border-right: 1px solid") %>% 
  column_spec(9, extra_css="border-right: 1px solid") %>%
  column_spec(10, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(11, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(12, background = "#F5F5F5", italic=TRUE) %>%
  column_spec(13, background = "#F5F5F5", italic=TRUE) %>%
  scroll_box(width = "100%", height = "300px")

```

$~$
$~$
$~$


We show how the number of significant SNPs differs for the 8 different scenarios with $S=0$ at the significance threshold $5 \times 10^{-8}$:

```{r, echo=FALSE}

results_nsig_5e_8_all <- read.csv("results/nsig_prop_bias_5e_8_LD_all.csv")


data_neutralS <- results_nsig_5e_8_all[which(results_nsig_5e_8_all$S==0),]
data_neutralS$scenario <- c(rep("Scenario 1",100), rep("Scenario 2",100), rep("Scenario 3",100), rep("Scenario 4",100), rep("Scenario 5",100), rep("Scenario 6",100), rep("Scenario 7",100), rep("Scenario 8",100))
data_neutralS$n_samples <- c(rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100), rep("30,000",100), rep("300,000",100))
data_neutralS <- data_neutralS[data_neutralS$n_sig != 0,]


ggplot(data_neutralS,aes(x=n_samples,y=n_sig,colour=scenario)) + geom_boxplot(position=position_dodge(0.2),size=0.5,aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + facet_grid(h2~prop_effect,labeller=label_both) + xlab("Sample size") + ylab("No. sig SNPs") + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())

```



We can also visualise the proportion of significant SNPs that are *significantly* overestimated and the MSE of significant SNPs at the significance threshold of $5 \times 10^{-8}$ as follows: 

```{r, echo=FALSE}
ggplot(data_neutralS,aes(x=n_samples,y=prop_x,colour=scenario)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + xlab("Sample size")+ facet_grid(h2~prop_effect,labeller=label_both) + ylab(expression(paste("Prop. sig SNPs", italic(" significantly "), "overestimated"))) + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())

ggplot(data_neutralS,aes(x=n_samples,y=mse,colour=scenario)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + xlab("Sample size")+ facet_grid(h2~prop_effect,labeller=label_both) + ylab("MSE of sig. SNPs") + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())


ggplot(data_neutralS,aes(x=n_sig,y=prop_x,colour=scenario)) + geom_point(aes(color=scenario)) +  scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7])) + xlab("No. sig SNPs")+ ylab(expression(paste("Prop. sig SNPs", italic(" significantly "), "overestimated"))) + theme_bw() + theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank())
```




## Application of Winner's Curse correction methods

Using the code detailed in `winnerscurse_sims_LD.R` and a total of 100 simulations, we evaluate the different Winner's Curse methods across each of the 8 scenarios. The methods to be compared include three conditional likelihood methods, FDR Inverse Quantile Transformation, the modified bootstrap method as well as five versions of the Empirical Bayes method. The following four bias evaluation metrics are used, with the first and second perhaps being the most important:

1.  The change in MSE of significant SNPs due to method implementation - `mse`
2.  The change in RMSE of significant SNPs due to method implementation - `rmse`
3.  The *relative* change in MSE of significant SNPs due to method implementation - `rel_mse`
4.  The fraction of significant SNPs in which their association estimates have been improved due to method implementation - `flb`

The formal mathematical definitions of these evaluation metrics may be found in the document which discussed simulating summary statistics of independent SNPs, with $E(\beta_i)$ replacing $\beta_i$.

We plot the results for the scenarios in which the selection parameter, S, is fixed at 0. The boxplots allow us to visualise the values obtained for the evaluation metric of interest in each of the 100 simulations. Four different rankings, one for each method, are also given. We first look at the case when the significance threshold is fixed at $5 \times 10^{-8}$ and then, follow this by analysing the results when the significance threshold is changed to a lower one of $5 \times 10^{-4}$.


**Evaluation metric 1 with threshold** $5 \times 10^{-8}$:

Summary of results for `mse` contained in `norm_5e-8_100sim_LD_ave.csv`:

```{r,echo=FALSE}

## summary results - mean! 
norm_5e_8_100sim <- read.csv("results/norm_5e-8_100sim_LD_ave.csv")

results_mse <- data.frame(Scenario=c(1:8),n_samples = c(rep(c("30,000","300,000"),4)), h2 = norm_5e_8_100sim$h2[1:8], prop_effect = norm_5e_8_100sim$prop_effect[1:8], S= norm_5e_8_100sim$S[1:8], EB = round(norm_5e_8_100sim$mse[1:8],6), EB_df = round(norm_5e_8_100sim$mse[(1+8):(8+8)],6), EB_scam = round(norm_5e_8_100sim$mse[(1+2*8):(8+2*8)],6), EB_gam_po = round(norm_5e_8_100sim$mse[(1+3*8):(8+3*8)],6), EB_gam_nb = round(norm_5e_8_100sim$mse[(1+4*8):(8+4*8)],6), FIQT = round(norm_5e_8_100sim$mse[(1+5*8):(8+5*8)],6), boot = round(norm_5e_8_100sim$mse[(1+6*8):(8+6*8)],6), cl1 = round(norm_5e_8_100sim$mse[(1+7*8):(8+7*8)],6), cl2 = round(norm_5e_8_100sim$mse[(1+8*8):(8+8*8)],6), cl3 = round(norm_5e_8_100sim$mse[(1+9*8):(8+9*8)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in MSE over all significant SNPs across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:15]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

## plotting full data set! 
norm_5e_8_100sim_all <- read.csv("results/norm_5e-8_100sim_LD_all.csv")

norm_5e_8_100sim_all$scenario <- c(rep(c(1:8),10))
norm_5e_8_100sim_all$scenario <- factor(norm_5e_8_100sim_all$scenario)


data_neutralS <- norm_5e_8_100sim_all[which(norm_5e_8_100sim_all$S==0),]


library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSa)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$



**Evaluation metric 2 with threshold** $5 \times 10^{-8}$:

Summary of results for `rmse` contained in `norm_5e-8_100sim_LD_ave.csv`:

```{r,echo=FALSE}
norm_5e_8_100sim <- read.csv("results/norm_5e-8_100sim_LD_ave.csv")
results_rmse <- data.frame(Scenario=c(1:8),n_samples = c(rep(c("30,000","300,000"),4)), h2 = norm_5e_8_100sim$h2[1:8], prop_effect = norm_5e_8_100sim$prop_effect[1:8], S= norm_5e_8_100sim$S[1:8], EB = round(norm_5e_8_100sim$rmse[1:8],6), EB_df = round(norm_5e_8_100sim$rmse[(1+8):(8+8)],6), EB_scam = round(norm_5e_8_100sim$rmse[(1+2*8):(8+2*8)],6), EB_gam_po = round(norm_5e_8_100sim$rmse[(1+3*8):(8+3*8)],6), EB_gam_nb = round(norm_5e_8_100sim$rmse[(1+4*8):(8+4*8)],6), FIQT = round(norm_5e_8_100sim$rmse[(1+5*8):(8+5*8)],6), boot = round(norm_5e_8_100sim$rmse[(1+6*8):(8+6*8)],6), cl1 = round(norm_5e_8_100sim$rmse[(1+7*8):(8+7*8)],6), cl2 = round(norm_5e_8_100sim$rmse[(1+8*8):(8+8*8)],6), cl3 = round(norm_5e_8_100sim$rmse[(1+9*8):(8+9*8)],6))

results_rmse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in RMSE over all significant SNPs across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_rmse[,6:15]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in RMSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}


data_neutralS <- norm_5e_8_100sim_all[which(norm_5e_8_100sim_all$S==0),]

data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSa)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=rmse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=rmse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$



**Evaluation metric 3 with threshold** $5 \times 10^{-8}$:

Summary of results for `rel_mse` contained in `norm_5e-8_100sim_LD_ave.csv`:

```{r,echo=FALSE}

results_rel_mse <- data.frame(Scenario=c(1:8),n_samples = c(rep(c("30,000","300,000"),4)), h2 = norm_5e_8_100sim$h2[1:8], prop_effect = norm_5e_8_100sim$prop_effect[1:8], S= norm_5e_8_100sim$S[1:8], EB = round(norm_5e_8_100sim$rel_mse[1:8],4), EB_df = round(norm_5e_8_100sim$rel_mse[(1+8):(8+8)],4), EB_scam = round(norm_5e_8_100sim$rel_mse[(1+2*8):(8+2*8)],4), EB_gam_po = round(norm_5e_8_100sim$rel_mse[(1+3*8):(8+3*8)],4), EB_gam_nb = round(norm_5e_8_100sim$rel_mse[(1+4*8):(8+4*8)],4), FIQT = round(norm_5e_8_100sim$rel_mse[(1+5*8):(8+5*8)],4), boot = round(norm_5e_8_100sim$rel_mse[(1+6*8):(8+6*8)],4), cl1 = round(norm_5e_8_100sim$rel_mse[(1+7*8):(8+7*8)],4), cl2 = round(norm_5e_8_100sim$rel_mse[(1+8*8):(8+8*8)],4), cl3 = round(norm_5e_8_100sim$rel_mse[(1+9*8):(8+9*8)],4))

results_rel_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#D4EFDF") %>%
  column_spec(2,background="#E9F7EF") %>%
  column_spec(3,background="#E9F7EF") %>%
  column_spec(4,background="#E9F7EF") %>%
  column_spec(5,background="#E9F7EF",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* relative change in MSE over all significant SNPs across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_rel_mse[,6:15]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Relative change* in MSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

ggplot(data_neutralSa,aes(x=n_samples,y=rel_mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
    ylab(expression(paste("Rel.", italic(" change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=rel_mse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
    ylab(expression(paste("Rel.", italic(" change "), "in MSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))

 
```

$~$ $~$ $~$ $~$

**Evaluation metric 4 with threshold** $5 \times 10^{-8}$:

Summary of results for `flb` contained in `norm_5e-8_100sim_LD_ave.csv`:

```{r,echo=FALSE}


results_flb <- data.frame(Scenario=c(1:8),n_samples = c(rep(c("30,000","300,000"),4)), h2 = norm_5e_8_100sim$h2[1:8], prop_effect = norm_5e_8_100sim$prop_effect[1:8], S= norm_5e_8_100sim$S[1:8], EB = round(norm_5e_8_100sim$flb[1:8],4), EB_df = round(norm_5e_8_100sim$flb[(1+8):(8+8)],4), EB_scam = round(norm_5e_8_100sim$flb[(1+2*8):(8+2*8)],4), EB_gam_po = round(norm_5e_8_100sim$flb[(1+3*8):(8+3*8)],4), EB_gam_nb = round(norm_5e_8_100sim$flb[(1+4*8):(8+4*8)],4), FIQT = round(norm_5e_8_100sim$flb[(1+5*8):(8+5*8)],4), boot = round(norm_5e_8_100sim$flb[(1+6*8):(8+6*8)],4), cl1 = round(norm_5e_8_100sim$flb[(1+7*8):(8+7*8)],4), cl2 = round(norm_5e_8_100sim$flb[(1+8*8):(8+8*8)],4), cl3 = round(norm_5e_8_100sim$flb[(1+9*8):(8+9*8)],4))


results_flb %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#D4E6F1") %>%
  column_spec(2,background="#EAF2F8") %>%
  column_spec(3,background="#EAF2F8") %>%
  column_spec(4,background="#EAF2F8") %>%
  column_spec(5,background="#EAF2F8",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")

```

$~$ $~$ $~$ $~$

The *mean* fraction of significant SNPs with improved association estimates due to method implementation across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_flb[,6:15]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in descending order:

```{r, echo=FALSE}
rank(-summary2)
```

$~$ $~$ $~$ $~$

Fraction of significant SNPs with improved association estimates due to method implementation, using a significance threshold of $5 \times 10^{-8}$:

```{r,echo=FALSE}

ggplot(data_neutralSa,aes(x=n_samples,y=flb,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
    ylab(expression(paste("Fraction of", italic(" improved "), "sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0.5, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=flb,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
    ylab(expression(paste("Fraction of", italic(" improved "), "sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0.5, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))



```

$~$ $~$ $~$ $~$

**Evaluation metric 2 with threshold** $5 \times 10^{-4}$:

Summary of results for `rmse` contained in `norm_5e-4_100sim_LD_ave.csv`:

```{r,echo=FALSE}
norm_5e_4_100sim <- read.csv("results/norm_5e-4_100sim_LD_ave.csv")
results_mse <- data.frame(Scenario=c(1:8),n_samples = c(rep(c("30,000","300,000"),4)), h2 = norm_5e_4_100sim$h2[1:8], prop_effect = norm_5e_4_100sim$prop_effect[1:8], S= norm_5e_4_100sim$S[1:8], EB = round(norm_5e_4_100sim$rmse[1:8],6), EB_df = round(norm_5e_4_100sim$rmse[(1+8):(8+8)],6), EB_scam = round(norm_5e_4_100sim$rmse[(1+2*8):(8+2*8)],6), EB_gam_po = round(norm_5e_4_100sim$rmse[(1+3*8):(8+3*8)],6), EB_gam_nb = round(norm_5e_4_100sim$rmse[(1+4*8):(8+4*8)],6), FIQT = round(norm_5e_4_100sim$rmse[(1+5*8):(8+5*8)],6), boot = round(norm_5e_4_100sim$rmse[(1+6*8):(8+6*8)],6), cl1 = round(norm_5e_4_100sim$rmse[(1+7*8):(8+7*8)],6), cl2 = round(norm_5e_4_100sim$rmse[(1+8*8):(8+8*8)],6), cl3 = round(norm_5e_4_100sim$rmse[(1+9*8):(8+9*8)],6))

results_mse %>%
kable() %>%
  kable_classic_2() %>%
kable_styling(bootstrap_options = "bordered",
full_width = FALSE,position="center") %>%
  row_spec(row = 0,bold = TRUE) %>%
  column_spec(1,background="#FCF3CF") %>%
  column_spec(2,background="#FEF9E7") %>%
  column_spec(3,background="#FEF9E7") %>%
  column_spec(4,background="#FEF9E7") %>%
  column_spec(5,background="#FEF9E7",extra_css="border-right: 1px solid") %>%
  scroll_box(width = "100%", height = "300px")
```

$~$ $~$ $~$ $~$

The *mean* change in RMSE over all significant SNPs across the 8 scenarios is obtained for each method:

```{r, echo=FALSE}
ranking <- results_mse[,6:15]
summary2 <- apply(ranking,2,mean,na.rm=TRUE)
summary2
```

The methods are ranked according to the results above in ascending order:

```{r, echo=FALSE}
rank(summary2)
```

$~$ $~$ $~$ $~$

*Change* in RMSE over all significant SNPs due to method implementation, using a significance threshold of $5 \times 10^{-4}$:

```{r,echo=FALSE}
## plotting full data set! 
norm_5e_4_100sim_all <- read.csv("results/norm_5e-4_100sim_LD_all.csv")


data_neutralS <- norm_5e_4_100sim_all[which(norm_5e_4_100sim_all$S==0),]


library(dplyr)
data_neutralS <- data_neutralS %>%
    mutate(method = recode(method, BR = 'boot' ))

data_neutralSa <- data_neutralS[which(data_neutralS$n_samples==30000),]
data_neutralSa$n_samples <- c(rep(c("30,000"),nrow(data_neutralSa)))

data_neutralSb <- data_neutralS[which(data_neutralS$n_samples==300000),]
data_neutralSb$n_samples <- c(rep(c("300,000"),nrow(data_neutralSa)))

## two plots, one for n_samples = 30,000 and one for n_samples = 300,000

ggplot(data_neutralSa,aes(x=n_samples,y=rmse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))


ggplot(data_neutralSb,aes(x=n_samples,y=rmse,colour=method)) + geom_boxplot(position=position_dodge(1),size=0.5,aes(color=method)) + facet_grid(h2~prop_effect,labeller=label_both) + scale_shape_manual(values=c(6,20,20,20,15,18,18,18,18,10)) + scale_color_manual(values=c(col[1],col[2],col[8],col[4],col[5],col[3],col[6],col[7],col1[1],col1[11])) + xlab("Sample size") + 
   ylab(expression(paste(italic("Change "), "in RMSE of sig. SNPs"))) + theme_bw() + geom_hline(yintercept=0, colour="black") +  theme(legend.title = element_blank(),legend.box.background = element_rect(colour = "black"),legend.spacing.y = unit(0, "mm"), legend.background=element_blank(), strip.text = element_text(face="italic"))
```

$~$ $~$ $~$ $~$


**Discussion:**

In our discussion of the results in this case, we focus on the first evaluation metric, namely 'change in MSE of significant SNPs' defined as above. At the $5 \times 10^{-8}$ significance threshold, we see the conditional likelihood methods perform poorly especially when sample sizes are 300,000. In most instances, they provide worse association estimates than the naïve approach. That said, it cannot be ignored that `cl2` seems to be one of the best adjustment methods when the proportion of effect SNPs is 0.01 and the sample size is 30,000. 

Interestingly, it appears that our proposed bootstrap method comes out on top overall here. In all situations depicted, it has one of the largest negative values for change in MSE of significant SNPs. FIQT tends to perform in a somewhat similar manner to this bootstrap method.

With respect to the empirical Bayes method and its variations, for sample sizes of 300,000, the most consistent forms are undeniably `EB`, `EB_gam_nb` and `EB_scam`. With the lower sample size, all versions seem to perform very similarly but with the original empirical Bayes method tending to have the smallest negative change in MSE when the proportion of effect SNPs is 0.01 and the heritability is 0.3. It is worth taking note here the considerable improvement in `EB_scam` when applied to this situation in which a simple correlation structure exists compared with the case of independent SNPs simulated previously. 

At the greater threshold of $5 \times 10^{-4}$, the first thing that comes to our attention is that positive values for change in MSE of significant SNPs are witnessed. This is in contrast to the previous simulation setting where the assumption of independence was present. There, at this threshold, all methods always led to better association estimates than the naive approach. The same was observed upon application of the methods to the UKBB real data sets. We will not place a great emphasis on the results at this threshold. That said, the most consistent and best performing methods here seem to be the bootstrap method, the original empirical Bayes method and the empirical Bayes method which uses shape constrained additive models (SCAMs).  



